### PR TITLE
Add triggerIdle function to manually trigger idle state - Fixes #503

### LINKED ISSUE #503: Add an options to turn on `isIdle` manually
### Describe The Problem To Be Solved

In my application when the screen is Idle I display a series of animations. Once `onActive` is called I stop those animations

I want to trigger those animations manually on some event but I still want the `onActive` function to stop those animations once the mouse moves (or any other event that triggers `isIdle` happens)

### Suggest A Solution

The easy solution for this would be a `triggerIdle` function which would be the opposite of `reset`

### PR DESCRIPTION
**Pull request by AyloSrd**
This pull request addresses the issue [#503](https://github.com/solidjs-community/solid-primitives/issues/503).

Currently, the `createIdleTimer` primitive does not provide a way to manually trigger the idle state. This pull request adds a new method called `triggerIdle` to the IdleTimer instance returned this primitive. The `triggerIdle` function sets the `isIdle` state to true, triggers the `onIdle` callback bypassing the prompt phase, and attaches listeners to the specified events that will call the `onActive` function if triggered. 

JSDoc and type definition were improved for better understanding and clarity.

cc. @itamarsch

### CODE REVIEW

File: packages/idle/src/types.ts [hunk 1]
```diff
@@ -2,21 +2,87 @@ import { Accessor } from "solid-js";
...
+   * @default () => {}
+   */
   onPrompt?: (promptEvt: Event) => void;
 }
 
-export interface IdleTimerReturn {
+/**
+ * The instance of the idle timer
+ */
+export interface IdleTimer {
```
Comment (thetarnav) [inline packages/idle/src/types.ts:63]: Can you add a `IdleTimerReturn` type that is an alias to `IdleTimer`, but with a `@deprecated` flag and some comment about a rename?

Comment (AyloSrd) [reply packages/idle/src/types.ts:63]: sure!

Comment (AyloSrd) [reply packages/idle/src/types.ts:63]: Done; please let me know if it's ok, and if further changes are needed.

**Comment by AyloSrd** [comment]
Forgot to update README, don't merge

**Comment by itamarsch** [comment]
Looks amazing! @thetarnav Can this be merged I really need it for an app of mine?

**Comment by thetarnav** [comment]
Yup, looks good, sorry about the delay

**Pull request merged**